File: bubble_sort.php
<?php
/**
 * @author: RunnerLee
 * @email: runnerleer@gmail.com
 * @time: 2018-05
 */

function bubble_sort(array $arr)
{
    $count = count($arr) - 1;

    for ($i = 0; $i < $count; ++$i) {
        for ($k = 0; $k < $count - $i; ++$k) {
            $temp = $arr[$k + 1];
            if ($arr[$k] > $temp) {
                $arr[$k + 1] = $arr[$k];
                $arr[$k] = $temp;
            }
        }
    }

    return $arr;
}

$arr = range(1, 10);
//
//$arr = [1,2,3,4,5,6,7,8,9,10,1,3,4,9,2];

shuffle($arr);

print_r($arr);

print_r(bubble_sort($arr));

File: bubble_sort.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int arr[10], i, k, temp;

    srand(0);

    for (i = 0; i < 10; i++) {
        arr[i] = rand();
    }

    for (i = 0; i < 8; i++) {
        for (k = 0; k < 8 - i; k++) {
            temp = arr[k + 1];
            if (arr[k] > temp) {
                arr[k + 1] = arr[k];
                arr[k] = temp;
            }
        }
    }

    for (i = 0; i < 9; i++) {
        printf("%d\n", arr[i]);
    }
}